plugin/federation/federation.go

```diff
@@ -243,100 +246,102 @@ func (f *federation) setEntities(schema *ast.Schema) {
 		}
 		if schemaType.Kind == ast.Object {
 			keys := schemaType.Directives.ForNames("key")
-			if len(keys) > 1 {
-				// TODO: support multiple keys -- multiple resolvers per Entity
-				panic("only one @key directive currently supported")
+			if len(keys) == 0 {
+				continue
 			}
 
-			if len(keys) > 0 {
-				dir := keys[0]
-				if len(dir.Arguments) != 1 || dir.Arguments[0].Name != "fields" {
-					panic("Exactly one `fields` argument needed for @key declaration.")
-				}
-				arg := dir.Arguments[0]
-				keyFieldSet := fieldset.New(arg.Value.Raw, nil)
-
-				// TODO: why is this nested inside the @key handling? -- because it's per-Entity, and we make one per @key
-				requires := []*Requires{}
-				for _, f := range schemaType.Fields {
-					dir := f.Directives.ForName("requires")
-					if dir == nil {
-						continue
-					}
-					requiresFieldSet := fieldset.New(dir.Arguments[0].Value.Raw, nil)
-					for _, field := range requiresFieldSet {
-						requires = append(requires, &Requires{
-							Name:  field.ToGoPrivate(),
-							Field: field,
-						})
+			e := &Entity{
+				Name:      schemaType.Name,
+				Def:       schemaType,
+				Resolvers: nil,
+				Requires:  nil,
+			}
+
+			// Let's process custom entity resolver settings.
+			dir := schemaType.Directives.ForName("entityResolver")
+			if dir != nil {
```

Comment:
Same sort of idea here.  Like a helper function like `configureEntityResolverDirective`...  And `configureKeyDirective`...  These are TODOs in my head.  Hoping to get more cycles for this over the break.

Reply:
Left this for you 🙂

Reply:
Thank you :)